src/test/groovy/org/cyclonedx/gradle/PluginConfigurationSpec.groovy

```diff
@@ -690,4 +691,77 @@ class PluginConfigurationSpec extends Specification {
         assert tool.getName() == "cyclonedx-gradle-plugin"
         assert tool.getVendor() == "CycloneDX"
     }
+
+    def "should include external reference - build-system"() {
+        given:
+        File testDir = TestUtils.createFromString("""
+            plugins {
+                id 'org.cyclonedx.bom'
+                id 'java'
+            }
+            repositories {
+                mavenCentral()
+            }
+            group = 'com.example'
+            version = '1.0.0'
+            cyclonedxBom {
+                includeBuildSystem = true
+            }
+            dependencies {
+            }""", "rootProject.name = 'hello-world'")
+
+        when:
+        def result = GradleRunner.create()
+            .withProjectDir(testDir)
+            .withEnvironment(["BUILD_URL" : "https://jenkins.example.com/job/123"])
+            .withArguments("cyclonedxBom", "--configuration-cache")
+            .withPluginClasspath()
+            .build()
+
+        then:
+        result.task(":cyclonedxBom").outcome == TaskOutcome.SUCCESS
+        File jsonBom = new File(testDir, "build/reports/bom.json")
+        Bom bom = new ObjectMapper().readValue(jsonBom, Bom.class)
+        assert bom.getMetadata().getComponent().getExternalReferences().size() == 1
+        ExternalReference buildSystemRef = bom.getMetadata().getComponent().getExternalReferences().get(0);
+        assert ExternalReference.Type.BUILD_SYSTEM == buildSystemRef.getType()
+        assert buildSystemRef.getUrl() == "https://jenkins.example.com/job/123"
+    }
+
+    def "should include external reference - build-system using configured environment variables"() {
+        given:
+        File testDir = TestUtils.createFromString('''
+            plugins {
+                id 'org.cyclonedx.bom'
+                id 'java'
+            }
+            repositories {
+                mavenCentral()
+            }
+            group = 'com.example'
+            version = '1.0.0'
+            cyclonedxBom {
+                includeBuildSystem = true
+                buildSystemEnvironmentVariable = '${SERVER}/build/${BUILD_ID}'
+            }
+            dependencies {
+            }''', "rootProject.name = 'hello-world'")
+
+        when:
+        def result = GradleRunner.create()
+            .withProjectDir(testDir)
+            .withEnvironment(["SERVER" : "https://ci.example.com", "BUILD_ID" : "123"])
```

Comment:
Please add tests when variable from the string does not exist, e.g. `buildSystemEnvironmentVariable = '${WRONG_NAME}'`

Reply:
done